aws_message/sqs.py

```diff
@@ -57,10 +65,18 @@ def __init__(self, sqs_settings):
             raise SQSException('No queue by name {}'.format(
                 self.queue_name))
 
-        if not re.match(r"^https://{}\.[^/]+/{}/{}$".format(
-                self.region, self.account_id, self.queue_name),
-                        self._queue.url):
-            raise SQSException('Invalid queue url {}'.format(self._queue.url))
+        if self.has_arn:
+            if not re.match(r"^https://{}\.[^/]+/{}/{}$".format(
+                    self.region, self.account_id, self.queue_name),
+                            self._queue.url):
+                raise SQSException('Invalid queue url {}'.format(
+                    self._queue.url))
+        else:
+            if not re.match(r"^https://queue.amazonaws.com/{}/{}$".format(
```

Comment:
Should this hostname be configurable?

Reply:
The queue URL is different from the topic_arn. 
We do not manually construct it, only verify it. The verification was in the earlier version and may not be necessary.